### DISCUSSION TITLE
Would it make sense for Submission.error to return a ZodError<Schema>

### DISCUSSION THREAD

**Discussion by swalker326** [edmundhung/conform#365] [General]
Hello, working with Conform has been amazing, thank you for all of the work. In working with it, I was wondering if the submission error key should return a ZodError that matches the api of the error response generate by zod. This would allow me to write 1 type of error handling and apply it to any parse (confrom parse or Schema.safeParse). It also brings up why the object is different at all. 

I think it makes sense to add fields, but when fields with the same key have different types like `error` or keys that have the same values have different keys `data` vs `value` it requires to writing different error handling logic for parsing errors from FormData vs errors from safeParse (I use this to validate objects)

```ts 
//Zod Schema.safeParse response
export declare type SafeParseSuccess<Output> = {
    success: true;
    data: Output;
};
export declare type SafeParseError<Input> = {
    success: false;
    error: ZodError<Input>;
};
```

```ts
//conform Submission response
export type Submission<Schema = any> = {
    intent: string;
    payload: Record<string, unknown>;
    error: Record<string, string[]>;
    value?: Schema | null;
};
```

An example use case from my remix projects. 

```ts
export async function action1({ request }: ActionFunctionArgs) {
  const userId = await requireUserId(request);
  const formData = await request.formData();
  const { error, value } = parse(formData, { schema: SignActionSchema });
  if (error || !value) {
    return json({ status: "error", error }, { status: 400 });
  }
  /*function to add the vote, handles all sorts of cases via Zod's Schema.safeParse
    But will always return: 
      status: "success" |"errors", 
        in error case
          error: ZodError<Schema>
        in success case
          data: Whatever data I need back on the client form useActionData
  */
  const response = await addVote({
    userId,
    videoId: value?.videoId,
    vote: value?.intent || "DOWNVOTE"
  });

  return json(response);
}
  ```
  
So in my `useActionData` I've built several different cases that I have to handle different data types in the same key or different keys for the same data type (the second case will never actually happen in the example. I never return the data from the value key of confroms parse but in other places I do).

**Comment by edmundhung** [comment]
Hi @swalker326. I am not sure if I follow your usage yet. Can you share an example on how you are handling the zod error object?
